columntypes/columntypes.go

```diff
@@ -0,0 +1,307 @@
+			cl = fmt.Sprintf("%s(%d,%d)", cl, t.ColumnLengths[key], *t.ColumnScales[key])
+		}
+
+		columns = append(columns, cl)
+	}
+
+	return strings.Join(columns, ",")
+}
+
+func isTypeWithRequiredLength(elem string) bool {
+	for _, val := range typesWithLength {
+		if val == elem {
+			return true
+		}
+	}
+
+	return false
+}
+
+// time layouts.
+var layouts = []string{
+	time.RFC3339, time.RFC3339Nano, time.Layout, time.ANSIC, time.UnixDate, time.RubyDate,
+	time.RFC822, time.RFC822Z, time.RFC850, time.RFC1123, time.RFC1123Z, time.RFC3339, time.RFC3339,
+	time.RFC3339Nano, time.Kitchen, time.Stamp, time.StampMilli, time.StampMicro, time.StampNano,
+}
+
+// Querier is a database querier interface needed for the GetColumnTypes function.
+type Querier interface {
+	QueryContext(ctx context.Context, query string, args ...any) (*sql.Rows, error)
+}
+
+// GetTableInfo returns a map containing all table's columns and their database types
+// and returns primary columns names.
+//
+//nolint:funlen,nolintlint
+func GetTableInfo(ctx context.Context, querier Querier, tableName string) (TableInfo, error) {
+	var primaryKeys []string
+
+	// check if table exist.
+	rows, err := querier.QueryContext(ctx, queryIfTableExist, tableName)
+	if err != nil {
+		return TableInfo{}, fmt.Errorf("execute query exist table: %w", err)
+	}
+
+	defer rows.Close() //nolint:staticcheck,nolintlint
+
+	for rows.Next() {
+		var count int
+		er := rows.Scan(&count)
+		if er != nil {
+			return TableInfo{}, fmt.Errorf("scan: %w", err)
+		}
+
+		if count == 0 {
+			return TableInfo{}, fmt.Errorf("table %s doesn't exist", tableName)
+		}
+	}
+
+	columnTypes := make(map[string]string)
+	columnLengths := make(map[string]int)
+	columnScales := make(map[string]*int)
+
+	rows, err = querier.QueryContext(ctx, querySchemaColumnTypes, strings.ToUpper(tableName))
+	if err != nil {
+		return TableInfo{}, fmt.Errorf("query get column types: %w", err)
+	}
+
+	for rows.Next() {
+		var (
+			columnName, dataType string
+			length               int
+			scale                *int
+		)
+
+		if er := rows.Scan(&columnName, &dataType, &length, &scale); er != nil {
+			return TableInfo{}, fmt.Errorf("scan rows: %w", er)
+		}
+
+		columnTypes[columnName] = dataType
+		columnLengths[columnName] = length
+		columnScales[columnName] = scale
+	}
+
+	rows, err = querier.QueryContext(ctx, queryGetPrimaryKeys, strings.ToUpper(tableName))
+	if err != nil {
+		return TableInfo{}, fmt.Errorf("query get column types: %w", err)
+	}
+
+	for rows.Next() {
+		var columnName string
+
+		if er := rows.Scan(&columnName); er != nil {
+			return TableInfo{}, fmt.Errorf("scan rows: %w", er)
+		}
+
+		primaryKeys = append(primaryKeys, columnName)
+	}
+
+	return TableInfo{
+		ColumnTypes:   columnTypes,
+		PrimaryKeys:   primaryKeys,
+		ColumnLengths: columnLengths,
+		ColumnScales:  columnScales,
+	}, nil
+}
+
+// ConvertStructureData converts a sdk.StructureData values to a proper database types.
+func ConvertStructureData(
```

Comment:
Tinies nitpick possible: `ConvertStructuredData` (missing `d` in `structured`)

Reply:
fixed